lib/utils/display.js

```diff
@@ -109,142 +128,213 @@ class Display {
   constructor ({ stdout, stderr }) {
     this.#stdout = setBlocking(stdout)
     this.#stderr = setBlocking(stderr)
+
+    // Handlers are set immediately so they can buffer all events
     process.on('log', this.#logHandler)
+    process.on('output', this.#outputHandler)
   }
 
   off () {
     process.off('log', this.#logHandler)
-    this.#logBuffer.length = 0
+    this.#logState.buffer.length = 0
+
+    process.off('output', this.#outputHandler)
+    this.#outputState.buffer.length = 0
+
     if (this.#progress) {
       this.#progress.stop()
     }
   }
 
-  load ({ loglevel, chalk, timing, unicode, progress, json, heading }) {
-    this.#chalk = chalk
-    this.#colors = COLOR_PALETTE({ chalk })
+  load ({
+    loglevel,
+    stdoutColor,
+    stdoutChalk,
+    stderrColor,
+    stderrChalk,
+    timing,
+    unicode,
+    progress,
+    json,
+    heading,
+  }) {
+    this.#stdoutColor = stdoutColor
+    this.#stdoutChalk = stdoutChalk
+
+    this.#stderrColor = stderrColor
+    this.#stderrChalk = stderrChalk
+
+    this.#logColors = COLOR_PALETTE({ chalk: stderrChalk })
 
     this.#levelIndex = LEVEL_OPTIONS[loglevel].index
     this.#timing = timing
     this.#json = json
     this.#heading = heading
 
+    // In silent mode we remove all the handlers
     if (this.#levelIndex <= 0) {
       this.off()
-    } else {
-      log.resume()
-      if (progress) {
-        this.#startProgress({ unicode })
-      }
+      return
     }
-  }
 
-  logTiming (...args) {
-    this.#logHandler(LEVELS.timing, ...args)
-  }
+    // Emit resume event on the logs which will flush output
+    log.resume()
 
-  forceLog (level, ...args) {
-    // This will show the log regardless of the current loglevel, except when silent
-    this.#logHandler({ level, force: true }, ...args)
-  }
+    // TODO: this should be a proc-log method `proc-log.output.flush`?
```

Comment:
Yes

Reply:
I'd like to handle these `flush` related TODOs in a follow-up PR.